proposals/bag-o-stacks/Explainer.md

```diff
@@ -217,21 +217,19 @@ If its stack reference operand is null or detached, `switch` traps. Otherwise, `
 ### `stack.new_switch` Create and switch to a new stack
 
 ```pseudo
-  C |- stack.new_switch x_1 y : t_1* -> t_2* rt
+  C |- stack.new_switch x_1 y : t_1* t_2* -> t_3* rt
   -- expand(C.TYPES[x_1]) = stack t_1* (ref null? x_2)
-  -- expand(C.FUNCS[y]) = func t_1* (ref null? x_2) -> []
-  -- expand(C.TYPES[x_2]) = stack t_2* rt
+  -- expand(C.FUNCS[y]) = func t_1* t_2* (ref null? x_2) -> []
+  -- expand(C.TYPES[x_2]) = stack t_3* rt
 ```
 
-`stack.new_switch` takes two immmediates: a type index `x_1` and a function index `y`. It is valid with type `t_1* -> t_2* rt` iff:
+`stack.new_switch` takes two immmediates: a type index `x_1` and a function index `y`. It is valid with type `t_1* t_2* -> t_3* rt` iff:
 
  - The expansion of the type at index `x_1` is a stack type with parameters `t_1* (ref null? x_2)`.
- - The expansion of the type of the function at index `y` is a function type `t_1* (ref null? x_2) -> []`
- - The expansion of the type at index `x_2` is a stack type with parameters `t_2* rt`
+ - The expansion of the type of the function at index `y` is a function type `t_1* t_2* (ref null? x_2) -> []`
+ - The expansion of the type at index `x_2` is a stack type with parameters `t_3* rt`
 
-`stack.new_switch x_1 y` both allocates a new stack and switches to it. It is equivalent to `(stack.new x y) (switch x)`, but engines should be able to implement it more efficiently because it calls the function immediately without having to stage the arguments anywhere.
```

Comment:
Do you mean 
`(stack.new y)(switch x)`
?
(Otherwise, there is an extraneous x floating around)

Reply:
No, because `stack.new` takes both the stack type index `x` and the function index `y`.

Reply:
got it.